### ISSUE TITLE
Remove dependency on Scarb's CASM output

### ISSUE THREAD

**Issue by Arcticae** [foundry-rs/starknet-foundry#1213]
Blocked by [1212](https://github.com/foundry-rs/starknet-foundry/issues/1212)

We should not use Scarb's CASM outputs since they are only for preview/debug purposes

**Comment by cptartur** [comment]
Duplicate of #1140
